crypto/x509util/crt.go

```diff
@@ -37,8 +39,12 @@ func SplitSANs(sans []string) (dnsNames []string, ips []net.IP, emails []string)
 		} else if ip := net.ParseIP(san); ip != nil {
 			ips = append(ips, ip)
 		} else {
-			// If not IP then assume DNSName.
-			dnsNames = append(dnsNames, san)
+			if u, err := url.Parse(san); err != nil || u.Scheme == "" {
+				// If not email, ip, or url, then dns name.
+				dnsNames = append(dnsNames, san)
+			} else {
+				uris = append(uris, u)
+			}
```

Comment:
This is probably not strictly ok, but I think it is good enough.

Reply:
yeah, I wasn't sure what the right answer was. 🤷‍♂️ If / when people complain, we'll fix it.